server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -309,15 +255,15 @@ void buildDiffAndSerializeStates() {
                 try {
                     if (sendFullVersion || previousState.nodes().nodeExists(node) == false) {
```

Comment:
Should we refactor `sendFullVersion` to `isFullVersion` or something along similar lines and pass this value when invoking `serializeClusterState(.., .., isFullVersion)` instead of hardcoding true/false. Will make the code more readable if similar changes can be done at other invocation of this method.

Reply:
In one of the conditions, if `sendFullVersion` is true, then also there is a case where we'll send diff. So, not planning to rely on this variable. Also during exception handling while sending cluster state, we need to send full cluster state and this variable doesn't contribute in that flow. So, eventually we'll anyway need to pass true/false manually.

Keeping it as it is.